data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverter.java

```diff
@@ -80,6 +80,7 @@ public void writeToBuffer(final AcknowledgementSet acknowledgementSet, List<Reco
         int eventCount = 0;
         for (Record record : records) {
             final long bytes = record.dynamodb().sizeBytes();
+            final Boolean userIdentity = record.userIdentity() != null && "dynamodb.amazonaws.com".equals(record.userIdentity().principalId());
```

Comment:
This variable should be named `isTtlDelete`, or the userIdentity could be passed to `addToBuffer` directly as a String like my comment above.

Also, would be good to have "dynamodb.amazonaws.com" as a private static variable whenever it is used.

Reply:
I moved the logic to `addToBuffer` but instead of passing a String, I was forced to pass the Identity as it also contains `Type` which was suggested for checking.